src/WPF/WPF.Viewer/Samples/Layers/AddVectorTiledLayerFromCustomStyle/AddVectorTiledLayerFromCustomStyle.xaml.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright 2022 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using Esri.ArcGISRuntime.Geometry;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Portal;
+using Esri.ArcGISRuntime.Tasks.Offline;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using System.Windows.Controls;
+
+namespace ArcGIS.WPF.Samples.AddVectorTiledLayerFromCustomStyle
+{
+    [ArcGIS.Samples.Shared.Attributes.Sample(
+        name: "Add vector tiled layer from custom style",
+        category: "Layers",
+        description: "Load ArcGIS vector tiled layers using custom styles.",
+        instructions: "Pan and zoom to explore the vector tile basemap.",
+        tags: new[] { "tiles", "vector", "vector basemap", "vector tiled layer", "vector tiles" })]
+    [ArcGIS.Samples.Shared.Attributes.OfflineData()]
+    public partial class AddVectorTiledLayerFromCustomStyle
+    {
+        // List all portal item IDs of vector tiled layers.
+        private List<string> _itemIDs;
```

Comment:
I think there should be two separate lists for item IDs. Using index to differentiate between between the items is bad practice. Sample can be broken by re-ordering the list, also involves a magic number with the index check.